src/transformers/image_processing_utils_fast.py

```diff
@@ -72,6 +78,25 @@
 logger = logging.get_logger(__name__)
 
 
+def get_best_available_device() -> "torch.device":
+    """
+    Get the best available torch device.
+    """
+    if is_torch_mlu_available():
+        return torch.device("mlu:0")
+    if is_torch_musa_available():
+        return torch.device("musa:0")
+    if is_torch_cuda_available():
+        return torch.device("cuda:0")
+    if is_torch_npu_available():
+        return torch.device("npu:0")
+    if is_torch_xpu_available(check_device=True):
+        return torch.device("xpu:0")
+    if is_torch_mps_available():
+        return torch.device("mps:0")
+    return torch.device("cpu")
```

Comment:
we could use `@lru_cache` here maybe, although I am not sure if we really gain performance but it doesn't hurt